privacyidea/lib/tokens/pushtoken.py

```diff
@@ -729,6 +729,32 @@ def create_challenge(self, transactionid=None, options=None):
                                                      challenge, fb_gateway,
                                                      pem_privkey, options)
             res = fb_gateway.submit_message(self.get_tokeninfo("firebase_token"), smartphone_data)
+
+            # Create the challenge in the challenge table regardless of a failure
+            # when sending the Push message to the Firebase API.
+            # But only if polling is allowed.
+            allow_polling = get_action_values_from_options(
+                SCOPE.AUTH, PUSH_ACTION.ALLOW_POLLING, options=options) or PushAllowPolling.ALLOW
+            if (allow_polling == PushAllowPolling.ALLOW or
```

Comment:
Maybe I am mislead here. But we also need to create a challenge in the database if the message is successfully submitted to firebase. Even if not polling is allowed. 
(Which would be the default behaviour. We sill need the challenge, in the database for the token to work correctly like before.)

So to my understanding this should be

~~~~python
if (polling-conditions) or (res):
~~~~

Reply:
You are absolutely right, good thing we have a review process :-)

Reply:
Done in cb4c8fb